### PR TITLE
endpoints with history of sign-ins from private IPs are accessed from…

### PR DESCRIPTION
**Pull request by timbMSFT**
… public IP address.

Fixes #

## Proposed Changes

  -
  -
  -

### CODE REVIEW

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿id: 4915c713-ab38-432e-800b-8e2d46933de6
+name: ssh newly internet-exposed endpoints
```
Comment (shainw) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: Probably want to capitalize the starting SSH here

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+description: endpoints with a history of sign-ins only from private IP addresses are accessed from a public IP address.
```
Comment (shainw) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: Capitalization on Endpoints, as this shows in the UX

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    dataTypes:
+      - Syslog
+queryFrequency: 24h
+queryPeriod: 7d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1078,T1110
```
Comment (shainw) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: Ask Sara, but I think she wants these on separate lines.

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| where privateAverage > 0
+// There is a non-zero number of logins from public IPs
+| where publicAverage >0
+// Approximate probability of seeing login from a public IP is < 1%
+| extend probabilityPublic = publicAverage / (privateAverage + publicAverage)
+| where probabilityPublic < 0.01
+// Today has the highest number of logins from public IPs that we've seen in the last week
+| extend publicLoginCountToday = publicIPLoginCountList[-1]
+| where publicLoginCountToday >= maxPublicLoginCount
+| extend HostCustomEntity = HostName
```
Comment (shainw) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: Looks like we can get a HostIP returned, add in the entity mapping for that also - extend IPCustomEntity = HostIP

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 5]
```diff
@@ -0,0 +1,57 @@
+﻿id: 4915c713-ab38-432e-800b-8e2d46933de6
+name: ssh newly internet-exposed endpoints
+description: endpoints with a history of sign-ins only from private IP addresses are accessed from a public IP address.
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Syslog
+    dataTypes:
+      - Syslog
+queryFrequency: 24h
```
Comment (shainw) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: Might make more sense to use 1d, just from a standardization perspective in Detections.  Not required.

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+relevantTechniques:
+  - T1078,T1110
+query: |
+
+let PrivateIPregex = @'^127\.|^10\.|^172\.1[6-9]\.|^172\.2[0-9]\.|^172\.3[0-1]\.|^192\.168\.'; 
+let ssh_logins = Syslog
+  | where TimeGenerated >= ago(7d)
+  | where Facility contains "auth" and ProcessName != "sudo"
+  | where SyslogMessage has "Accepted"
+  | extend SourceIP = extract("(([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9])\\.([0-9]|[0-9][0-9]|[0-9][0-9][0-9]))",1,SyslogMessage)
```
Comment (petebryan) [inline Detections/Syslog/ssh_NewlyInternetExposed.yaml]: If you want to make things a bit more readable you can adjust this regex to:
(([0-9]{1,3})\\.([0-9]{1,3})\\.([0-9]{1,3})\\.(([0-9]{1,3})))

I've used the expanded one in the past due to issues with { in queries defined in Notebooks, but I think in these queries this format is a bit neater.

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml (hunk 2 above)
Comment (timbMSFT) [reply Detections/Syslog/ssh_NewlyInternetExposed.yaml]: done

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml (hunk 3 above)
Comment (timbMSFT) [reply Detections/Syslog/ssh_NewlyInternetExposed.yaml]: done

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml (hunk 5 above)
Comment (timbMSFT) [reply Detections/Syslog/ssh_NewlyInternetExposed.yaml]: done

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml (hunk 6 above)
Comment (timbMSFT) [reply Detections/Syslog/ssh_NewlyInternetExposed.yaml]: done

File: Detections/Syslog/ssh_NewlyInternetExposed.yaml (hunk 4 above)
Comment (timbMSFT) [reply Detections/Syslog/ssh_NewlyInternetExposed.yaml]: the IPs get aggregated so there is not an IP column in the results (just a field reflecting the count of distinct private versus public IPs)

**Pull request merged**